src/wifi.rs

```diff
@@ -1442,3 +1526,90 @@ impl WifiWait {
         }
     }
 }
+
+#[cfg(feature = "async")]
+pub struct AsyncWifiDriver<'d, T>
+where
+    T: BorrowMut<WifiDriver<'d>>,
+{
+    base_driver: T,
+    _p: PhantomData<WifiDriver<'d>>,
+}
+
+#[cfg(feature = "async")]
+impl<'d, T> AsyncWifiDriver<'d, T>
+where
+    T: BorrowMut<WifiDriver<'d>>,
+{
+    #[cfg(all(feature = "alloc", esp_idf_comp_nvs_flash_enabled))]
+    pub fn new(base_driver: T) -> Result<Self, EspError> {
```

Comment:
As per above, move all async functions from `WifiDriver` to `AsyncWifiDriver` and turn them into plain ol `pub async fn` variants, and strip their `_async` suffix.